week08/movie-app/src/pages/MainPage.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import { Link } from "react-router-dom";
+import Movie from "../components/Movie";
+import Header from "../components/Header";
+import axios from "axios";
+
+const Title = styled.div`
+    font-family: 'HSSanTokki20-Regular';
+    font-size: 40px;
+    margin-top: 20px;
+    margin-bottom: 40px;
+`;
+
+const MainContainer = styled.div`
+    display: flex;
+    justify-content: center;
+    flex-direction: column;
+    align-items: center;
+`;
+
+const Container = styled.div`
+    background-color: none;
+`;
+
+const MovieList = styled.div`
+    display: flex;
+    justify-content: center;
+    flex-direction: row;
+    flex-wrap: wrap;
+`;
+
+const StyledLink = styled(Link)`
+    color: #FFFFFF;
+    text-decoration-line: none;
+`;
+
+const MainPage = () => {
+    const [movies, setMovies] = useState([]);
+    const [sort, setSort] = useState("");
+    const [genre, setGenre] = useState("");
+    const [title, setTitle] = useState("영화 목록");
+
+    useEffect(() => {
+        async function getMovies() {
+            try {
+                const genreQuery = genre ? `&genre=${genre}` : "";
+                const res = await axios.get(
+                    `https://yts.mx/api/v2/list_movies.json?sort_by=${sort}${genreQuery}`
+                );
+                setMovies(res.data.data.movies);
+            } catch (error) {
+                console.log(error);
+            }
+        }
+        getMovies();
+    }, [sort, genre]);

```

Comment:
P5: useEffect 내부에서 비동기 함수를 바로 호출해도 좋지만, 외부에서 선언한 뒤에 사용하면 코드의 가독성이 높아지고 함수의 재사용을 높일 수 있습니다. 아래는 제가 작성해본 코드입니다. 
```suggestion
const getMovies = async (sort, genre) => {
  try {
    const res = await axios.get(
      `https://yts.mx/api/v2/list_movies.json?sort_by=${sort}${genre ? `&genre=${genre}` : ""}`
    );
    setMovies(res.data.data.movies);
  } catch (error) {
    console.log(error);
  }
};

useEffect(() => {
  getMovies(sort, genre);
}, [sort, genre]);
```